src/test/clone/test_fork.rs

```diff
@@ -220,6 +222,168 @@ fn test_child_change_group() -> Result<(), Box<dyn Error>> {
     Ok(())
 }
 
+/// Helper to run the given test function in a child process. This is helpful to
+/// avoid cross-test interference. e.g. `f` can manipulate signal handlers and
+/// masks without having to restore them, and will only have child processes
+/// that it spawns itself.
+///
+/// Returns `Ok(())` if `f` completes without panicking, and an error if `f`
+/// panicked.
+fn run_test_in_subprocess(f: impl FnOnce()) -> anyhow::Result<()> {
+    let (reader, writer) = rustix::pipe::pipe().unwrap();
+    let clone_res = unsafe { linux_api::sched::fork() }.unwrap();
+    let _child_pid = match clone_res {
+        CloneResult::CallerIsChild => {
+            // Ensure we exit with non-zero exit code on panic.
+            std::panic::set_hook(Box::new(|info| {
+                eprintln!("panic: {info:?}");
+                unsafe { libc::exit(1) };
+            }));
+            f();
+            assert_eq!(rustix::io::write(&writer, &[0]), Ok(1));
+            unsafe { libc::exit(0) };
+        }
+        CloneResult::CallerIsParent(child_pid) => child_pid,
+    };
+
+    // Close our copy of the writer-end of the pipe, so that parent hang trying
+    // to read from the pipe if the child exited abnormally.
+    drop(writer);
+
+    // Because waitpid isn't implemented yet, we get the "exit code"
+    // from a pipe.
+    // TODO: once waitpid is implemented, use that instead.
+    let mut exit_code = [0xff_u8];
+    ensure_ord!(rustix::io::read(&reader, &mut exit_code), ==, Ok(1));
+    ensure_ord!(exit_code[0], ==, 0);
+
+    Ok(())
+}
+
+fn test_exit_signal_normal_exit(exit_signal: nix::sys::signal::Signal) -> anyhow::Result<()> {
+    run_test_in_subprocess(|| {
+        static mut SIGNO: c_int = 0;
+        static mut INFO: MaybeUninit<siginfo_t> = MaybeUninit::uninit();
```

Comment:
Why not `Option` to avoid any chance of UB (I guess other than the UB inherent in `static mut` if the signal handler writes to it while the test is in the middle of reading it)?

Reply:
Done